config/element-api.php

```diff
@@ -221,6 +221,81 @@ function extractCaseStudySummary($entry)
     ];
 }
 
+/**
+ * Looks up an old version or draft of an entry
+ * @usage: `list('entry' => $entry, 'status' => $status) = getDraftOrVersionOfEntry($entry);`
+ */
+function getDraftOrVersionOfEntry($entry)
+{
+    $isDraft = \Craft::$app->request->getParam('draft');
+    $isVersion = \Craft::$app->request->getParam('version');
+
+    if ($isDraft) {
+        $status = 'draft';
+        $revisionId = $isDraft;
+        $revisionMethod = 'getDraftsByEntryId';
+        $entryRevisionMethod = 'getDraftById';
+        $revisionIdParam = 'draftId';
+    } else if ($isVersion) {
+        $status = 'version';
+        $revisionId = $isVersion;
+        $revisionMethod = 'getVersionsByEntryId';
+        $entryRevisionMethod = 'getVersionById';
+        $revisionIdParam = 'versionId';
+    }
+
+    if (($isDraft || $isVersion) && $revisionId) {
+
+        // Get all drafts/revisions of this post
+        $revisions = \Craft::$app->entryRevisions->{$revisionMethod}($entry->id, $entry->siteId);
+
+        // Filter drafts/revisions for the requested ID
+        $revisions = array_filter($revisions, function ($revision) use ($revisionId, $revisionIdParam, $entryRevisionMethod) {
+            return $revision->{$revisionIdParam} == $revisionId;
+        });
+
+        // Is this draft/revision ID valid for this post?
+        if (count($revisions) > 0) {
+
+            // Look up the revision itself
+            $revision = \Craft::$app->entryRevisions->{$entryRevisionMethod}($revisionId);
+
+            if ($revision) {
+                // Non-live content has a null URI in Craft,
+                // so restore it to its base entry's URI
+                $revision->uri = $entry->uri;
+                return [
+                    'entry' => $revision,
+                    'status' => $status,
+                ];
+            }
+        }
+    }
+
+    // default to the original, unmodified entry
+    return [
+        'entry' => $entry,
+        'status' => $entry->status,
+    ];
+}
+
+function getAvailableLanguages($locale, $section, $slug) {
+    $alternateSite = $locale === 'en' ? 'cy' : 'en';
```

Comment:
Could probably simplify this to:

`$alternateSite = $locale === 'en' ? 'cy' : $locale;`

(less repetition of `en` string)